src/main/java/org/opentripplanner/openstreetmap/wayproperty/WayPropertySet.java

```diff
@@ -202,7 +205,8 @@ public I18NString getCreativeNameForWay(OSMWithTags way) {
     if (bestNamer == null) {
       return null;
     }
-    return bestNamer.generateCreativeName(way);
+    var name = bestNamer.generateCreativeName(way);
+    return deduplicator.deduplicateObject(I18NString.class, name);
```

Comment:
Here you are deduplicating in the caller end, not in the callee end where the creation is happening - this of cause lead to potentially less DRY code since the deduplication could be done in one place, not for every caller. Might not be a problem here - but it is better to be consistent. So, when we have Di in place, maybe we should revisit this an inject the deduplicator into the "I18NString" factories instead. Pushing it 2 levels inn where it belong: `WayPropertySet -> Namer -> I18NString` factory.